### PR TITLE
Add prefsync

### PR DESCRIPTION
**Pull request by asmeurer**
It's a little tool that I wrote. Just testing this out, to see how it works.

This tool is pure Python, so it will technically work anywhere, but it only makes sense on OS X. How can I disable Linux and Windows?

### CODE REVIEW

**Comment by asmeurer** [comment]
So the link style from the example recipe doesn't work?

**Comment by jakirkham** [comment]
The `/s/` part of the URL should be `/p/`.

To disable non-OS X builds just add `skip: true # [not osx]` to the `build` section.

Sorry on my phone. So if this is incomprehensible that is why.

File: recipes/prefsync/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+{% set version = "1.2" %}
+{% set sha256 = "66f0d3870b442e715de23b632e83f27f5971885b40bb3f6b211437138c554496" %}
+
+package:
+  name: prefsync
+  version: {{ version }}
+
+source:
+  fn: prefsync-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/s/prefsync/prefsync-{{ version }}.tar.gz
```
Comment (jakirkham) [inline recipes/prefsync/meta.yaml]: To make my terse phone comment more clear this should be...

```
https://pypi.io/packages/source/p/prefsync/prefsync-{{ version }}.tar.gz
```

File: recipes/prefsync/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
```
Comment (jakirkham) [inline recipes/prefsync/meta.yaml]: Also to clarify my terse phone comment, this should be as follows to build on OS X only.

```
  skip: true  # [not osx]
```

File: recipes/prefsync/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  home: https://github.com/asmeurer/prefsync
+  license: MIT
+  summary: |
+    A little tool to help synchronize Mac OS X plist files (used for
+    preferences for most Mac Apps) seamlessly, in a way that can be tracked by
+    git.
+  description: |
+    A little tool to help synchronize Mac OS X plist files (used for
+    preferences for most Mac Apps) seamlessly, in a way that can be tracked by
+    git.
```
Comment (jakirkham) [inline recipes/prefsync/meta.yaml]: So, my understanding from what has been relayed to me is the `summary` should be a quick one liner. The `description` can be a bit longer.

File: recipes/prefsync/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  doc_url: https://github.com/asmeurer/prefsync
+  dev_url: https://github.com/asmeurer/prefsync
```
Comment (jakirkham) [inline recipes/prefsync/meta.yaml]: Feel free to keep these if they help you. Though they are unenforced ATM.

**Comment by jakirkham** [comment]
Looks pretty reasonable to me. Some minor things to address.

File: recipes/prefsync/meta.yaml (hunk 1 above)
Comment (asmeurer) [inline recipes/prefsync/meta.yaml]: OK. I just took that from the example recipe in this repo.

File: recipes/prefsync/meta.yaml (hunk 3 above)
Comment (asmeurer) [inline recipes/prefsync/meta.yaml]: So should I copy the full [readme](https://github.com/asmeurer/prefsync/) to the description?

File: recipes/prefsync/meta.yaml (hunk 4 above)
Comment (asmeurer) [inline recipes/prefsync/meta.yaml]: If I had more time I would create a real website and docs for this. Probably won't at this point in time, though.

File: recipes/prefsync/meta.yaml (hunk 1 above)
Comment (asmeurer) [inline recipes/prefsync/meta.yaml]: Oh I get it. The `s` is the first letter of `simplejson`.

Comment (jakirkham) [inline recipes/prefsync/meta.yaml]: Yeah, I made a similar error when I first saw the new URLs. The `s` based example might be an unfortunate choice as it is not too unreasonable to assume it stands for `source` or similar. I've been making my PyPI recipes like this [one]( https://github.com/conda-forge/funcsigs-feedstock/blob/f896b21228dd6dcaa74c2c60cad9b3f1b4648ebb/recipe/meta.yaml ) to keep life simple. Maybe we should change the example to match.

File: recipes/prefsync/meta.yaml (hunk 3 above)
Comment (jakirkham) [inline recipes/prefsync/meta.yaml]: That's probably overkill. Honestly it is up to you whether you want the `description` or not. I'm guessing it is only supposed to be a few sentences. Though we really haven't decided if we want to use it or really how to. While the example has it, only `home`, `license`, `summary`, and, in some cases, `license_file` are enforced ATM. The rest are technically optional. See this issue ( https://github.com/conda-forge/conda-forge.github.io/issues/119 ) for more details

File: recipes/prefsync/meta.yaml (hunk 4 above)
Comment (jakirkham) [inline recipes/prefsync/meta.yaml]: Basically the same comment above w.r.t. to the `description` applies here.

File: recipes/prefsync/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+source:
+  fn: prefsync-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/p/prefsync/prefsync-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: true  # [not osx]
+  script: python setup.py install --single-version-externally-managed --record record.txt
```
Comment (jakirkham) [inline recipes/prefsync/meta.yaml]: As you are not using `setuptools`, please just make this `python setup.py install` with no extra arguments.

File: recipes/prefsync/meta.yaml (hunk 3 above)
Comment (asmeurer) [inline recipes/prefsync/meta.yaml]: I can delete it, as it duplicates summary. I'm not really clear where either one is used. 

By the way, suggestion: mention which fields are optional in the example. The README makes it sound like all recipes should look like the example, and I know you have a linter so I didn't want to go to awry.

File: recipes/prefsync/meta.yaml (hunk 4 above)
Comment (asmeurer) [inline recipes/prefsync/meta.yaml]: To be sure, that is also the dev url, and it is where the "docs" live. If anything, this project doesn't have a home url.

File: recipes/prefsync/meta.yaml (hunk 3 above)
Comment (jakirkham) [inline recipes/prefsync/meta.yaml]: Deletion of the `description` is fine by me. The linter won't complain and if it does I have your back. 😄

`summary` shows up in the Readme. Here's an [example]( https://github.com/conda-forge/sympy-feedstock/blob/1d3ff23f1e46c4e9edc91acd50d347d5064844a8/README.md#about-sympy ).

I think @pelson planned for the `description` to go there too, but there is a bug that needs to be fixed first ( https://github.com/conda-forge/conda-smithy/pull/227 ). So, it doesn't work currently.

Comment (jakirkham) [inline recipes/prefsync/meta.yaml]: > By the way, suggestion: mention which fields are optional in the example. The README makes it sound like all recipes should look like the example, and I know you have a linter so I didn't want to go to awry.

Noted.

File: recipes/prefsync/meta.yaml (hunk 4 above)
Comment (jakirkham) [inline recipes/prefsync/meta.yaml]: Yep, again feel free to keep or delete these as you see fit. I'm happy with either decision. The linter won't complain.

File: recipes/prefsync/meta.yaml (hunk 3 above)
Comment (jakirkham) [inline recipes/prefsync/meta.yaml]: I guess there is this [comment]( https://github.com/conda-forge/staged-recipes/blob/55997067f51258d59121033ece13dabc210fc155/recipes/example/meta.yaml#L42 ). Are you suggesting a similar comment be in the Readme?

Comment (asmeurer) [inline recipes/prefsync/meta.yaml]: Oh, so it is mentioned. That's enough. I guess I just added them anyway.

File: recipes/prefsync/meta.yaml (hunk 5 above)
Comment (asmeurer) [inline recipes/prefsync/meta.yaml]: Oh right :)

**Comment by jakirkham** [comment]
LGTM

**Pull request merged**